nixos/modules/services/misc/radicle.nix

```diff
@@ -0,0 +1,515 @@
+            };
+            peers = {
+              type = mkOption {
+                type = types.enum [ "dynamic" "static" ];
+                default = "dynamic";
+                description = ''
+                  "dynamic" for a dynamic peer set
+
+                  "static" for a static peer set.
+                  Connect to the configured peers and maintain the connections.
+                '';
+              };
+              target = mkOption {
+                type = types.ints.unsigned;
+                default = 8;
+                description = "Number of outbound dynamic peers";
+              };
+            };
+            policy = mkOption {
+              type = types.enum [ "allow" "block" ];
+              default = "block";
+              description = "Default seeding policy";
+            };
+            proxy = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              description = "Global proxy";
+            };
+            relay = mkOption {
+              type = types.enum [ "auto" "always" "never" ];
+              default = "never";
+              description = "Whether or not our node should relay messages";
+            };
+            scope = mkOption {
+              type = types.enum [ "all" "followed" ];
+              default = "all";
+              description = "Default seeding scope";
+            };
+            workers = mkOption {
+              type = types.ints.positive;
+              default = 8;
+              description = "Number of worker threads to spawn";
+            };
+          };
+          options.preferredSeeds = mkOption {
+            type = with types; listOf str;
+            default = [
+              "z6MkrLMMsiPWUcNPHcRajuMi9mDfYckSoJyPwwnknocNYPm7@seed.radicle.garden:8776"
+              "z6Mkmqogy2qEM2ummccUthFEaaHvyYmYBYh3dbe9W4ebScxo@ash.radicle.garden:8776"
+            ];
+            description = ''
+              Preferred seeds. These seeds will be used for explorer links
+              and in other situations when a seed needs to be chosen
+            '';
+          };
+          options.publicExplorer = mkOption {
+            type = types.str;
+            default = "https://app.radicle.xyz/nodes/$host/$rid$path";
+            description = ''
+              Public explorer. This is used for generating links
+            '';
+          };
+          options.web = {
+            pinned = {
+              repositories = mkOption {
+                type = with types; listOf str;
+                default = [];
+                description = ''
+                  Pinned repositories on a Web client.
+                '';
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services = let
+      commonConfig = serviceName:
+        {
+          environment = env // {
+            RUST_BACKTRACE = "1";
+            RUST_LOG = mkDefault "info";
```

Comment:
is this necessary?

Reply:
No, I don't think it's necessary, I took it from https://app.radicle.xyz/nodes/seed.radicle.garden/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5/tree/systemd/radicle-node.service#L24 and left it in case some hardening options were causing troubles, but I guess it does not belong to a production environment.
I'll remove [RUST_BACKTRACE](https://doc.rust-lang.org/std/backtrace/), I can always set it on my personal config if need be.